rosbag2_storage/src/rosbag2_storage/metadata_io.cpp

```diff
@@ -39,6 +39,24 @@
 
 namespace YAML
 {
+/// Allow pass a metadata version to the sub-structs of BagMetadata for deserializing.
+/**
+  * Encoding should always use the current metadata version, so it does not need this value.
+  * We cannot extend the YAML::Node class to include this, so we must call it
+  * as a function with the node as an argument.
+  */
+template<typename T>
+T versioned_decode(const Node & node, int version)
+{
+  if (!node.IsDefined()) {
+    throw TypedBadConversion<T>(node.Mark());
+  }
+  T t;
```

Comment:
Is allocating `T` here OK performance wise? `t` is not a good variable name, what can this represent? Also allocating it here instead of passing it in forces `T` to be default constructible. Not sure if this could be an issue.

Reply:
It's the logic that yaml-cpp uses, I'd prefer to not get too far from it. Can rename the typevar. But yes, allocating T is fine. This is a fixed cost at startup, it doesn't affect runtime performance, and the total amount of data involved is small even for complex bags, a few KB at most